jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,211 @@
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.AsUndirectedGraph;
+import org.jgrapht.traverse.LexicographicalBfsIterator;
+import org.jgrapht.traverse.MaximumCardinalityIterator;
+
+import java.util.*;
+
+/**
+ * Implementation of the lexicographical breadth-first search algorithm for chordal graph recognition.
+ * <p>
+ * A <a href="https://en.wikipedia.org/wiki/Chordal_graph">chordal graph</a> is one in which all cycles of four or more vertices have a chord, which is an edge that
+ * is not part of the cycle but connects two vertices of the cycle.
+ * <p>
+ * A graph is chordal iff its the vertices can be arranged into a perfect elimination order. Perfect elimination
+ * order isn't unique. Both maximum cardinality search and lexicographical breadth-first search produces this order.
+ * As a result, chordality inspection can be performed in two way. Here it is done by default via MCS,
+ * because it outperforms LexBFS by a constant factor in practice. Nevertheless, lexicographical breadth-first
+ * order can be obtained via {@link ChordalityInspector#getLexicographicalBfsOrder()}.
+ * <p>
+ * Both lexicographical BFS and maximum cardinality search run in O(|V| + |E|). Checking whether given order is the perfect elimination order
+ * via {@link ChordalityInspector#isPerfectEliminationOrder(List)} takes O(|V| + |E|) as well. So,
+ * overall time complexity of the method {@link ChordalityInspector#isChordal()} is O(|V| + |E|).
+ *
+ * @param <V> the graph vertex type.
+ * @param <E> the graph edge type.
+ * @author Timofey Chudakov
+ * @see LexicographicalBfsIterator
+ * @see MaximumCardinalityIterator
+ */
+public class ChordalityInspector<V, E> {
+    /**
+     * The inspected graph.
+     */
+    private Graph<V, E> graph;
+    /**
+     * Contains true if the graph is chordal, otherwise false. Is null before the first call to the
+     * {@link ChordalityInspector#isChordal()}.
+     */
+    private Boolean chordal = null;
+
+    /**
+     * Vertices of the {@code graph} in a maximum cardinality order.
+     */
+    private List<V> mcsOrder;
+    /**
+     * Vertices of the {@code graph} in a lexicographical breadth-first order.
+     */
+    private List<V> lexBfsOrder;
+
+
+    /**
+     * Creates a chordality inspector for {@code graph}.
+     *
+     * @param graph the graph for which a chordality inspector to be created.
+     */
+    public ChordalityInspector(Graph<V, E> graph) {
```

Comment:
I would prefer the class to have a single method 'isChordal' and people can change the behavior by the constructor. This means that you should be able to adjust the behavior of the class to either use LexBFS or MCS.